packages/custom-storage/src/sql/SQLStorage.ts

```diff
@@ -390,7 +401,8 @@ export class SQLStorage {
     visId: string,
     datasets: Array<Dataset|string>,
     overwriteDatasets: boolean,
-    isPrivateVis: boolean) {
+    isPrivateVis: boolean,
```

Comment:
Maybe is the moment to change some functions params to:
`functionName(mandatoryParam1, mandatoryParam2, {opts})`
to be able to add new params easier.
In this case:
`uploadAndLinkDatasetsTo(visId, datasets, {overwriteDatasets, isPrivateVis, event})`

Reply:
I thought about it, and yes, I completely agree with you on that refactor. It will allow us also to avoid having to fill some other 'optional' methods

Reply:
I have just added all events as optionals params inside generic Options objects, in case you want to have a look at it